src/file_pipe_log/pipe_builder.rs

```diff
@@ -87,6 +90,129 @@ impl<F: FileSystem> DualPipesBuilder<F> {
         }
     }
 
+    /// Manually reads through log items in all available log files of the
+    /// specified queue, and replays them to specific [`ReplayMachine`]s
+    /// that can be constructed via `machine_factory`.
+    pub fn recover_queue<M: ReplayMachine, FA: Factory<M>>(
```

Comment:
Could you keep the position unchanged? Previously it reflected the order of method calling (scan -> recover -> recover_queue).

Reply:
Done